### DISCUSSION TITLE
How to throw appropriately on subscribe call?

### DISCUSSION THREAD

**Discussion by theogravity** [hayes/pothos#195] [Q&A] [answered]
I have a service with multiple subscription handlers, and would like to be able to throw in certain circumstances (eg subscription-specific perm check) when the handler is called. I'm able to do this in our non-giraphql infrastructure, but it seems I can only do this in the resolver definition in giraph, which means that an event has to pass through first before I can throw.

(I want you to know that I am using merged schemas, so it could be something specific to a merged schema and the native behavior actually works.)

```
gqlBuilder.subscriptionField('events', (t) => {
  return t.field({
    type: BaseEvent,
    subscribe: () => {
       throw new Error('test') // "iterator.next is not a function" gets thrown to the client instead of the error
    },
    resolve: () => {
       throw new Error('test') // this comes down appropriately, but an event has to be processed first before it hits this
    }
 })   
```

**Comment by hayes** [comment, marked as answer]
GiraphQL passes subscribe and resolve directly through to the underlying schema (some plugins do some wrapping in certain cases) but generally the subscribe and resolve functions are exactly what ends up in the built schema.  This is likely an issue with how the schemas are merged.

Hard to say for sure without more details

**Comment by chris-parker-switchboard** [comment]
Got to this discussion through a link in our code that Theo, the original reporter, put there.  Tested this again in the version that we're using, 3.10.1, and it appears that this now works as we expect.
